File: ObjectiveC-DI/ObjectiveC-DI/ConsoleLogger.m
//
//  ConsoleLogger.m
//  ObjectiveC-DI
//
//  Created by Ilya Puchka on 28.02.16.
//  Copyright © 2016 Ilay Puchka. All rights reserved.
//

#import "ConsoleLogger.h"

@implementation ConsoleLogger

- (void)log:(NSString *)message {
    NSLog(@"[%@ <%p>]: %@", [self class], self, message);
}

@end

File: Swift-DI/Swift-DI/ConsoleLogger.swift
//
//  ConsoleLogger.swift
//  Swift-DI
//
//  Created by Ilya Puchka on 28.02.16.
//  Copyright © 2016 Ilay Puchka. All rights reserved.
//

import Foundation

class ConsoleLogger: NSObject, Logger {
    func log(message: String) {
        print("[\(self.dynamicType) <\(unsafeAddressOf(self))>]: \(message)")
    }
}
